# HG changeset patch
# User Rik <rik@octave.org>
# Date 1551552978 28800
#      Sat Mar 02 10:56:18 2019 -0800
# Node ID 4eed5ba2586a9bc16efe83ab7c78a5619b26647f
# Parent  2703e1407887f352f296161ad76f72b6fe35304d
legend.m: Remove obsolete specification of "location" by single numeric arg.

* legend.m: Remove FIXME and obsolete input checking for a numeric position
specifier as final argument to legend function.

diff --git a/scripts/plot/appearance/legend.m b/scripts/plot/appearance/legend.m
--- a/scripts/plot/appearance/legend.m
+++ b/scripts/plot/appearance/legend.m
@@ -198,33 +198,16 @@ function [hleg, hleg_obj, hplot, labels]
   location = "default";
   show = "create";
   textpos = "default";
   box = "default";
   delete_leg = false;
   find_leg_hdl = (nargs == 0);  # possibly overridden
   propvals = {};
 
-  ## Process old way of specifying location with a number rather than a string.
-  ## FIXME: When can this be deleted?
-  if (nargs > 0)
-    pos = varargin{nargs};
-    if (isnumeric (pos) && isscalar (pos) && pos == fix (pos))
-      warning ("legend: specifying location with a numeric argument is obsolete and will be removed from a future version of Octave, use a string specification instead");
-      if (pos >= -1 && pos <= 4)
-        location = [{"northeastoutside", "best", "northeast",
-                     "northwest", "southwest", "southeast"}] {pos + 2};
-        varargin(nargs) = [];
-        nargs -= 1;
-      else
-        error ("legend: invalid location specified");
-      endif
-    endif
-  endif
-
   ## Find "location" and "orientation" property/value pairs
   while (nargs > 1)
     pos = varargin{nargs-1};
     str = varargin{nargs};
     if (strcmpi (pos, "location") && ischar (str))
       location = lower (str);
       varargin(nargs-1:nargs) = [];
       nargs -= 2;
